Formula/c/chatgpt-cli.rb

```diff
@@ -0,0 +1,40 @@
+class ChatgptCli < Formula
+  desc "A CLI for the OpenAI ChatGPT API"
+  homepage "https://github.com/kardolus/chatgpt-cli"
+  version "1.5.5"
+
+  if Hardware::CPU.intel?
+    url "https://github.com/kardolus/chatgpt-cli/releases/download/v#{version}/chatgpt-darwin-amd64"
+    sha256 "ba290456b9c6dd99da9f12a5a87b3a684774bf501c3a436abddaba89be3a186f"
+  elsif Hardware::CPU.arm?
+    url "https://github.com/kardolus/chatgpt-cli/releases/download/v#{version}/chatgpt-darwin-arm64"
+    sha256 "61ff45c8408e8b1900f39dda7034e9cf0991f816a0414fdc4447ec9dbcdd88ba"
+  end
+
+  def install
+    if Hardware::CPU.intel?
+        bin.install "chatgpt-darwin-amd64" => "chatgpt"
+    elsif Hardware::CPU.arm?
+        bin.install "chatgpt-darwin-arm64" => "chatgpt"
+    end
+  end
+
+  test do
+    system "#{bin}/chatgpt", "--help"
+  end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hmm, this is an interesting point. I have unit, integration and contract tests in my repo: https://github.com/kardolus/chatgpt-cli -- testing the full lifecycle here: https://github.com/kardolus/chatgpt-cli/blob/422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0/integration/integration_test.go#L265

However, in order to talk to chatgpt you need an API key. 

I did improve the test though to check the keys found in the configuration. Does that work?